src/parse.jl

```diff
@@ -10,12 +10,39 @@ function init_dates_extension()
     )
 end
 
-const ISOZonedDateTimeFormat = let
+begin
+    # Needs to be initialized to construct formats
     init_dates_extension()
-    DateFormat("yyyy-mm-ddTHH:MM:SS.ssszzz")
+
+    # Follows the ISO 8601 standard for date and time with an offset. See
+    # `Dates.ISODateTimeFormat` for the `DateTime` equivalent.
+    const ISOZonedDateTimeFormat = DateFormat("yyyy-mm-dd\\THH:MM:SS.ssszzz")
+    const ISOZonedDateTimeNoMillisecondFormat = DateFormat("yyyy-mm-dd\\THH:MM:SSzzz")
+end
+
+function Base.parse(::Type{ZonedDateTime}, str::AbstractString)
+    # Works as the format should only contain a period when milliseconds are included
+    return if contains(str, '.')
+        parse(ZonedDateTime, str, ISOZonedDateTimeFormat)
+    else
+        parse(ZonedDateTime, str, ISOZonedDateTimeNoMillisecondFormat)
+    end
 end
 
-Dates.default_format(::Type{ZonedDateTime}) = ISOZonedDateTimeFormat
```

Comment:
As you said this isn't documented anywhere so it's technically not public. I think it's safest to deprecate this in the off chance anyone is using this.